pkg/sdk/go/groups.go

```diff
@@ -0,0 +1,232 @@
+	if err != nil {
+		return "", err
+	}
+
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusCreated {
+		return "", errors.Wrap(ErrFailedCreation, errors.New(resp.Status))
+	}
+
+	id := strings.TrimPrefix(resp.Header.Get("Location"), fmt.Sprintf("/%s/", groupsEndpoint))
+	return id, nil
+}
+
+func (sdk mfSDK) DeleteUsersGroup(id, token string) error {
+	endpoint := fmt.Sprintf("%s/%s", groupsEndpoint, id)
+
+	url := createURL(sdk.baseURL, sdk.groupsPrefix, endpoint)
+
+	req, err := http.NewRequest(http.MethodDelete, url, nil)
+	if err != nil {
+		return err
+	}
+
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusNoContent {
+		return errors.Wrap(ErrFailedRemoval, errors.New(resp.Status))
+	}
+
+	return nil
+
+}
+
+func (sdk mfSDK) AssignUserToGroup(userID, groupID, token string) error {
+	endpoint := fmt.Sprintf("%s/%s/users/%s", groupsEndpoint, groupID, userID)
+	url := createURL(sdk.baseURL, sdk.groupsPrefix, endpoint)
+	req, err := http.NewRequest(http.MethodPut, url, bytes.NewReader([]byte{}))
+	if err != nil {
+		return err
+	}
+
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return errors.Wrap(ErrFailedConnect, errors.New(resp.Status))
+	}
+
+	return nil
+}
+
+func (sdk mfSDK) RemoveUserFromGroup(userID, groupID, token string) error {
+	endpoint := fmt.Sprintf("%s/%s/users/%s", groupsEndpoint, groupID, userID)
+	url := createURL(sdk.baseURL, sdk.groupsPrefix, endpoint)
+
+	req, err := http.NewRequest(http.MethodDelete, url, bytes.NewReader([]byte{}))
+	if err != nil {
+		return err
+	}
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return err
+	}
+
+	if resp.StatusCode != http.StatusNoContent {
+		return errors.Wrap(ErrFailedRemoval, errors.New(resp.Status))
+	}
+
+	return nil
+}
+
+func (sdk mfSDK) ListUsersForGroup(groupID, token string, offset, limit uint64) (UsersPage, error) {
+	endpoint := fmt.Sprintf("%s/%s/users?offset=%d&limit=%d&", groupsEndpoint, groupID, offset, limit)
+	url := createURL(sdk.baseURL, sdk.groupsPrefix, endpoint)
+
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return UsersPage{}, err
+	}
+
+	resp, err := sdk.sendRequest(req, token, string(CTJSON))
+	if err != nil {
+		return UsersPage{}, err
+	}
+	defer resp.Body.Close()
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return UsersPage{}, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return UsersPage{}, errors.Wrap(ErrFailedFetch, errors.New(resp.Status))
+	}
+
+	var tp UsersPage
+	if err := json.Unmarshal(body, &tp); err != nil {
+		return UsersPage{}, err
+	}
+
+	return tp, nil
+}
+
+func (sdk mfSDK) UsersGroups(token string, offset, limit uint64, id string) (UsersGroupsPage, error) {
```

Comment:
Didi you meant `UserGroups` ?

Reply:
I'm not sure what would fit better here, this method will list Groups, later we will introduce groups for things so we will have ThingsGroups, if I use `User` instead of `Users` may sound like list Groups for certain user